packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet.tsx

```diff
@@ -0,0 +1,138 @@
+import {Component, h, Prop, State} from '@stencil/core';
+import {
+  InitializableComponent,
+  InitializeBindings,
+  Bindings,
+  BindStateToController,
+} from '../../utils/initialization-utils';
+import {
+  buildSmartSnippet,
+  SmartSnippet,
+  SmartSnippetState,
+  ResultTemplatesHelpers,
+  buildInteractiveResult,
+} from '@coveo/headless';
+import {Hidden} from '../common/hidden';
+import {Heading} from '../common/heading';
+import {LinkWithResultAnalytics} from '../result-link/result-link';
+
+/**
+ * The `atomic-smart-snippet` component displays the excerpt of a document that would be most likely to answer a particular query.
+ *
+ * @part smart-snippet - The wrapper of the entire smart snippet.
+ * @part heading - The header displaying the question that is answered by the found document excerpt.
+ * @part answer - The found document excerpt.
+ * @part source-url - The URL to the document the excerpt is from.
+ * @part source-title - The title of the document the excerpt is from.
+ * @part footer - The footer underneath the answer.
+ */
+@Component({
+  tag: 'atomic-smart-snippet',
+  styleUrl: 'atomic-smart-snippet.pcss',
+  shadow: true,
+})
+export class AtomicSmartSnippet implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  public smartSnippet!: SmartSnippet;
+  @BindStateToController('smartSnippet')
+  @State()
+  public smartSnippetState!: SmartSnippetState;
+  public error!: Error;
+
+  /**
+   * The [heading level](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements) to use for the question at the top of the snippet, from 1 to 6.
+   *
+   * It's strongly recommended to set this property in order to improve accessibility.
+   */
+  @Prop({reflect: true}) public headingLevel = 0;
+
+  public initialize() {
+    this.smartSnippet = buildSmartSnippet(this.bindings.engine);
+  }
+
+  public get source() {
+    if (!this.smartSnippetState.answerFound) {
+      return null;
+    }
+    const {contentIdKey, contentIdValue} = this.smartSnippetState.documentId;
+    const linkedDocument = this.bindings.engine.state.search.results.find(
```

Comment:
We should add this logic to the headless controller. Probably as a computed state property on the `smartSnippet` controller.

Reply:
Will do in a separate PR then